demos/kotlin-app/src/main/kotlin/com/pusher/chatkitdemo/MainActivity.kt

```diff
@@ -48,22 +46,10 @@ class MainActivity : AppCompatActivity() {
         state = State.Idle
 
         launch {
-            app.events.map { it as? ErrorOccurred }
-                .filterNotNull()
-                .consumeEach { (error) ->
-                    state = State.Failed(error)
-                }
-        }
-
-        launch {
-            val roomService = app.rooms()
-            state = roomService
-                .flatMap { it.fetchUserRooms(true).await() }
-                .fold({ error ->
-                    State.Failed(error)
-                }, { rooms ->
-                    State.Loaded(rooms)
-                })
+            state = app.rooms()
+                .flatMap { it.fetchUserRooms().await() }
+                .map {rooms -> rooms.filter { it.memberUserIds.size < 100 } }
```

Comment:
Why do you need this check here?

Reply:
This is part of the app logic, before it was done in the presentation (which it actually still is and will remove). This will go into a `viewModel` so that the presentation layer shouldn't care about this.